qa/rustfmt.sh

```diff
@@ -0,0 +1,15 @@
+#! /bin/sh
+
+r=0
+# check the list of known rustfmt-ed files
+cat qa/rustfmt.txt | while read i; do
+    cargo fmt $i;
+    # the file used to be formatted and is not anymore
+    if [ $(git diff -- $i | wc -l) -gt 0 ]; then
```

Comment:
Nothing wrong with this check, just FYI.

An alternative to check for differences would be `git diff --quiet -- $i` which makes `git diff` "exit with codes similar to `diff`". I.e. returns 1 if there are any differences, 0 otherwise. See `git help diff`

Reply:
Thanks, looks nicer.

What is the exact bash syntax for the test ?

Reply:
Different ways to skin that cat.

You might have seen something like:
```bash
git diff --quiet -- $i
if [ $? -ne 0 ] ; then
   # multiple commands ...
fi
```
I'd probably would do (if you don't need the exit value of the command):
```bash
if ! git diff --quiet -- $i; then
   # multiple commands ...
fi
```
Note the inversion with `! ...` as the `then` path is executed if the command returns 0. `diff`/`git diff --quiet` returns 1 (not 0 basically) if files are different.

Or you could short-circuit things
```bash
git diff --quiet -- $i || ( 
   # multiple commands ...
)
```

Or if you fancy one liners - not my personal preference if more than one command in failure path, but valid as well:
```bash
git diff --quiet -- $i || ( git diff -- $i; echo "$i needs formatting"; r=1 )
```

It's really your personal preference